applications/ChimeraApplication/custom_processes/apply_chimera_process.cpp

```diff
@@ -0,0 +1,627 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+// ==============================================================================
+//  ChimeraApplication
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Authors:        Aditya Ghantasala, https://github.com/adityaghantasala
+// 					Navaneeth K Narayanan
+//					Rishith Ellath Meethal
+// ==============================================================================
+//
+
+// Application includes
+#include "custom_processes/apply_chimera_process.h"
+#include "containers/model.h"
+#include "utilities/builtin_timer.h"
+#include "utilities/variable_utils.h"
+
+namespace Kratos {
+
+template <int TDim>
+ApplyChimera<TDim>::ApplyChimera(ModelPart& rMainModelPart, Parameters iParameters)
+    : mrMainModelPart(rMainModelPart), mParameters(iParameters)
+{
+    // This is only for example
+    Parameters example_parameters(R"(
+            {
+               	"chimera_parts"   :   [
+									[{
+										"model_part_name":"PLEASE_SPECIFY",
+                                        "search_model_part_name":"PLEASE_SPECIFY",
+                                        "boundary_model_part_name":"PLEASE_SPECIFY",
+										"overlap_distance":0.0
+									}],
+									[{
+										"model_part_name":"PLEASE_SPECIFY",
+                                        "search_model_part_name":"PLEASE_SPECIFY",
+                                        "boundary_model_part_name":"PLEASE_SPECIFY",
+										"overlap_distance":0.0
+									}],
+									[{
+										"model_part_name":"PLEASE_SPECIFY",
+                                        "search_model_part_name":"PLEASE_SPECIFY",
+                                        "boundary_model_part_name":"PLEASE_SPECIFY",
+										"overlap_distance":0.0
+									}]
+								]
+            })");
+    mNumberOfLevels = mParameters.size();
+    KRATOS_ERROR_IF(mNumberOfLevels < 2)
+        << "Chimera requires atleast one Patch !" << std::endl;
+
+    mEchoLevel = 0;
+    mReformulateEveryStep = false;
+    mIsFormulated = false;
+}
+
+template <int TDim>
+ApplyChimera<TDim>::~ApplyChimera()
+{
+}
```

Comment:
This can be removed and implemented in the header as `~ApplyChimera() = default`